tests/platform_tests/api/test_module.py

```diff
@@ -108,10 +108,10 @@ def test_get_presence(self, duthosts, enum_rand_one_per_hwsku_hostname, localhos
             if self.expect(presence is not None, "Unable to retrieve module {} presence".format(i)):
                 if self.expect(isinstance(presence, bool), "Module {} presence appears incorrect".format(i)):
                     name = module.get_name(platform_api_conn, i)
-                    if name in self.skip_mod_list:
-                        self.expect(presence is False, "Module {} is not present".format(i))
-                    else:
+                    if name not in self.skip_mod_list:
                         self.expect(presence is True, "Module {} is not present".format(i))
+                    else:
```

Comment:
What is the expected state of 'presence' in else?

Reply:
It depends on the specific vendor implementation.
One vendor implemented this API as following:

    def get_presence(self):
        if self.get_oper_status() == ModuleBase.MODULE_STATUS_EMPTY:
            return False
        else:
            return True

So most likely all vendors implemented this way and for the logical chassis issue that we observed (see the issue https://github.com/sonic-net/sonic-mgmt/issues/6160 I filed), although the LC is marked to be skipped but it is physically plugged in and "Online" so the get_presence will return "True" even if it is marked to be skipped over.

The thing is if the inventory file specified the module is to be skipped, the test code should not try to guess what it must be. it could be plugged in and admin shut or it could be in our case online for another "logical chassis" but not for the one that we are running the test for...

Reply:
so which means that, if module is not in skip list, we should expect presence to be "true", if the module is in skip list, then we should not expect any specific value for 'presence'